analyses/molecular-subtyping-pathology/02-incorporate-clinical-feedback.Rmd

```diff
@@ -0,0 +1,200 @@
+# Assigning params$is_ci to running_in_ci avoids a locked binding error
+running_in_ci <- params$is_ci
+
+# Are we testing? In case of a non 0/1 number, we recast as logical, and then 
+# ensure logical.
+if (running_in_ci %in% c(0,1)) running_in_ci <- as.logical(running_in_ci)
+if (!(is.logical(running_in_ci)))
+{
+  stop("\n\nERROR: The parameter `is_ci` should be FALSE/TRUE (or 0/1).")
+}
+```
+
+
+### Directory and files
+
+#### Directories
+
+```{r}
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+results_dir <- "results"
+```
+
+
+#### Input
+
+When we run this locally, we want to tie it to a specific version of the histologies file _prior_ to reviewed clinical subtypes for PNOC003 samples and pathology feedback [#751](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/751)
+
+```{r}
+if (running_in_ci) {
+  # clinical pbta histology
+clinical <- readr::read_tsv(file.path(data_dir, 
+                                "pbta-histologies.tsv")) 
+} else {
+    # clinical pbta histology
+clinical <- readr::read_tsv(file.path(data_dir, "release-v15-20200228",
+                                "pbta-histologies.tsv")) 
+}
+```
+
+
+```{r}
+library("tidyverse")
+
+# clinical review file for PNOC003 
+pnoc003_review <- readr::read_tsv(file.path("input","pnoc003-clinical-review.tsv"))
+  
+# subtyping from molecular-subtype-HGG
+subtype <- readr::read_tsv(file.path(results_dir,
+                         "compiled_molecular_subtypes.tsv")) %>%  
+  # ids to match
+  left_join(clinical[,c("Kids_First_Participant_ID",
+                        "Kids_First_Biospecimen_ID",
+                        # need tumor_descriptor to match to clinical review
+                        "tumor_descriptor")],by=c("Kids_First_Biospecimen_ID",
+                                       "Kids_First_Participant_ID")) 
+
+```
+
+#### Output file
+
+```{r}
+output_file <- file.path(results_dir, "compiled_molecular_subtypes_with_clinical_feedback.tsv")
+```
+
+
+Identify pnoc003 samples that have differences between clinical review subtype
+and molecular subtype from histone variant mutation status
+
+```{r}
+diff_subtype <- subtype %>% left_join(pnoc003_review,by=c("Kids_First_Participant_ID","tumor_descriptor")) %>% 
+  dplyr::filter(
+    # checking for samples where clinically reviewed subtype is H3 WT
+    # but molecular subtype is not HGG
+    (grepl("H3 WT",.$`H3 Status`) & !grepl("HGG",molecular_subtype)) |
+    # checking for samples where clinically reviewed subtype is H3 K28M
+    # but molecular subtype is not DMG H3 K28
+    (grepl("K28M",.$`H3 Status`) & !grepl("DMG, H3 K28",molecular_subtype)) | 
+    # checking for samples were clinical review was not found  
+    grepl("no report",.$`H3 Status`) )  %>%
+  dplyr::select(Kids_First_Participant_ID,
+                sample_id,
+                `H3 Status`,
+                molecular_subtype,
+                tumor_descriptor)
```

Comment:
It looks like there are essentially duplicates in `diff_subtype`. I expect this is because `compiled_molecular_subtypes.tsv` is laid out with one biospecimen per row and you're joining by participant identifier and `tumor_descriptor`. So I'm thinking it could be useful to include a `dplyr::distinct()` step somewhere in here? Maybe just for the display below.